src/vivarium_gates_nutrition_optimization/components/pregnancy.py

```diff
@@ -158,50 +208,27 @@ def get_initial_event_times(self, pop_data: SimulantData) -> pd.DataFrame:
 
 
 class PregnancyModel(DiseaseModel):
-    def setup(self, builder: Builder):
-        """Perform this component's setup."""
-        super(DiseaseModel, self).setup(builder)
-
-        self.configuration_age_start = builder.configuration.population.age_start
-        self.configuration_age_end = builder.configuration.population.age_end
-
-        cause_specific_mortality_rate = self.load_cause_specific_mortality_rate_data(builder)
-        self.cause_specific_mortality_rate = builder.lookup.build_table(
-            cause_specific_mortality_rate,
-            key_columns=["sex"],
-            parameter_columns=["age", "year"],
-        )
-        builder.value.register_value_modifier(
-            "cause_specific_mortality_rate",
-            self.adjust_cause_specific_mortality_rate,
-            requires_columns=["age", "sex"],
-        )
-
-        self.population_view = builder.population.get_view(["age", "sex", self.state_column])
-        builder.population.initializes_simulants(
-            self.on_initialize_simulants,
-            creates_columns=[self.state_column],
-            requires_columns=["age", "sex"],
-            requires_streams=[f"{self.state_column}_initial_states"],
-        )
-        self.randomness = builder.randomness.get_stream(f"{self.state_column}_initial_states")
-
-        builder.event.register_listener("time_step", self.on_time_step, priority=3)
-        builder.event.register_listener("time_step__cleanup", self.on_time_step_cleanup)
+    @property
+    def time_step_priority(self) -> int:
+        return 3
 
 
 def Pregnancy():
     not_pregnant = NotPregnantState(models.PREGNANT_STATE_NAME)
     pregnant = PregnantState(
         models.PREGNANT_STATE_NAME,
+        allow_self_transition=True,
         get_data_functions={
             "prevalence": lambda *_: 1.0,
             "disability_weight": lambda *_: 0.0,
             "excess_mortality_rate": lambda *_: 0.0,
+            # Add a dummy dwell time so we can overwrite it later
+            "dwell_time": lambda *_: None,
```

Comment:
Can you remind me how this gets overwritten?

Reply:
We set the dwell time with get_dwell_time_pipeline, which here pulls from the population view, not the get_data_functions as in the superclass.